doc/development.rst

```diff
@@ -172,3 +172,186 @@ If there are open questions, final decisions are made by the Temporary
+
+.. code-block:: RST
+
+    .. _vx.y.z:
+
+    x.y.z
+    =====
+    **Released MONTH YEAR**
+
+    HIGHLIGHTS
+    ----------
+    
+    - Nilearn now includes functionality A
+    - ...
+
+
+Next, we need to bump the version number of Nilearn by updating the file `nilearn/version.py` with the new version number, that is edit the line:
+
+.. code-block:: python
+
+    __version__ = x.y.z.dev
+
+
+to be:
+
+.. code-block:: python
+   
+    __version__ = x.y.z
+
+
+We also need to update the website news section by editing the file `nilearn/doc/themes/nilearn/layout.html`. The news section typically contains links to the last 3 releases that should look like:
+
+.. code-block:: html
+    
+    <h4> News </h4>
+        <ul>
+            <li><p><strong>November 2020</strong>: 
+                <a href="whats_new.html#v0-7-0">Nilearn 0.7.0 released</a>
+            </p></li>
+            <li><p><strong>February 2020</strong>: 
+                <a href="whats_new.html#v0-6-2">Nilearn 0.6.2 released</a>
+            </p></li>
+            <li><p><strong>January 2020</strong>: 
+                <a href="whats_new.html#v0-6-1">Nilearn 0.6.1 released</a>
+            </p></li>
+        </ul>
+
+
+Here, we should remove the last entry and add the new release on top of the list.
+
+In addition, we can have a look at `MANIFEST.in` to check that all additional files that we want to be included or excluded from the release are indicated. Normally we shouldn't have to touch this file.
+
+Add these changes and the version tag, and commit them:
+
+.. code:: bash
+    
+    git add doc/whats_new.rst nilearn.version.py
+    git commit -m "REL x.y.z"
+    git tag x.y.z
+    git push origin release-x.y.z
+    git push --tags
+
+
+Build the distributions and upload them to Pypi
+-----------------------------------------------
+
+First of all we should make sure we don't include files that shouldn't be present:
+
+.. code-block:: bash
+    
+    git checkout x.y.z
+
+
+If the workspace contains a `dist` folder, make sure to clean it:
+
+.. code-block:: bash
+    
+    rm -r dist
+
+
+In order to build the binary wheel files, we need to install `wheel <https://pypi.org/project/wheel/>`_:
+
+.. code-block:: bash
+    
+    pip install wheel
+
+
+And, in order to upload to `Pypi`, we will use `twine <https://pypi.org/project/twine/>`_ that you can also install with `pip`:
+
+.. code-block:: bash
+    
+    pip install twine
+
+
+Build the source and binary distributions:
+
+.. code-block:: bash
+    
+    python setup.py sdist bdist_wheel
+
+
+This should add two files to the `dist` subfolder:
+
+- one for the source distribution that should look like `PACKAGENAME-VERSION.tar.gz`
+- one for the binary distribution that should look like `PACKAGENAME-PACKAGEVERSION-PYTHONVERSION-PYTHONCVERSION-PLATFORM.whl`
```

Comment:
maybe the term "binary distribution" is a bit confusing, as it is just source code + installation instructions in a compressed archive -- no compiled binaries

Reply:
see for example https://packaging.python.org/glossary/#glossary